transforms/class.js

```diff
@@ -105,11 +108,14 @@ module.exports = (file, api, options) => {
     return !invalidProperties.length;
   };
 
-  const hasMixins = classPath => {
-    if (ReactUtils.hasMixins(classPath)) {
+  const hasInconvertibleMixins = classPath => {
+    if (
+      ReactUtils.hasMixins(classPath) &&
+      !ReactUtils.hasSpecificMixins(classPath, ['ReactComponentWithPureRenderMixin'])
```

Comment:
For open source, it’s important that we support `PureRenderMixin` here. This is how we refer to it [in the docs](https://facebook.github.io/react/docs/pure-render-mixin.html).